odoo/addons/base/models/res_partner.py

```diff
@@ -756,61 +756,44 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
-            self._apply_ir_rules(where_query, 'read')
-            from_clause, where_clause, where_clause_params = where_query.get_sql()
-            from_str = from_clause if from_clause else 'res_partner'
-            where_str = where_clause and (" WHERE %s AND " % where_clause) or ' WHERE '
-
-            # search on the name of the contacts and of its company
-            search_name = name
-            if operator in ('ilike', 'like'):
-                search_name = '%%%s%%' % name
-            if operator in ('=ilike', '=like'):
-                operator = operator[1:]
+            self = self.with_user(name_get_uid) if name_get_uid else self
 
             unaccent = get_unaccent_wrapper(self.env.cr)
+            domain = AND([args, self._get_name_search_domain(operator, name)])
 
-            fields = self._get_name_search_order_by_fields()
-
-            query = """SELECT res_partner.id
-                         FROM {from_str}
-                      {where} ({email} {operator} {percent}
-                           OR {display_name} {operator} {percent}
-                           OR {reference} {operator} {percent}
-                           OR {vat} {operator} {percent})
-                           -- don't panic, trust postgres bitmap
-                     ORDER BY {fields} {display_name} {operator} {percent} desc,
-                              {display_name}
-                    """.format(from_str=from_str,
-                               fields=fields,
-                               where=where_str,
-                               operator=operator,
-                               email=unaccent('res_partner.email'),
-                               display_name=unaccent('res_partner.display_name'),
-                               reference=unaccent('res_partner.ref'),
-                               percent=unaccent('%s'),
-                               vat=unaccent('res_partner.vat'),)
-
-            where_clause_params += [search_name]*3  # for email / display_name, reference
-            where_clause_params += [re.sub('[^a-zA-Z0-9\-\.]+', '', search_name) or None]  # for vat
-            where_clause_params += [search_name]  # for order by
-            if limit:
-                query += ' limit %s'
-                where_clause_params.append(limit)
-            self.env.cr.execute(query, where_clause_params)
-            return [row[0] for row in self.env.cr.fetchall()]
+            # don't use _search() because it can be return list of ids
+            self.check_access_rights('read')
+            self._flush_search(domain)
+            query = self._where_calc(domain)
+            self._apply_ir_rules(query, 'read')
+            if limit is not None:
+                query.limit = int(limit)
+
+            order = [self._get_name_search_order_by_fields()]
+            if name:
+                search_name = name
+                if operator in ('ilike', 'like'):
+                    search_name = f'%%{name}%%'
```

Comment:
The wildcard character for operator `LIKE` is `%`.  The double `%s%s` was escaping it for %-formatting the string with `name`.  As we use an f-string instead of a %-formatted string, we no longer need to double it.
```suggestion
                    search_name = f'%{name}%'
```